earn/src/components/borrow/modal/RepayModal.tsx

```diff
@@ -136,7 +136,10 @@ function RepayButton(props: RepayButtonProps) {
 
   // MARK: Preparing data that's necessary to figure out button state -------------------------------------------------
   const existingLiability = marginAccount.liabilities[lender === marginAccount.lender0 ? 'amount0' : 'amount1'];
-  const bigExistingLiability = ethers.utils.parseUnits(existingLiability.toString(), repayToken.decimals);
+  const bigExistingLiability = ethers.utils.parseUnits(
+    removeScientificNotation(existingLiability.toString(), repayToken.decimals),
+    repayToken.decimals
+  );
```

Comment:
I'm not sure this is the right solution. Instead, I think we should just do `ethers.utils.parseUnits(existingLiability.toFixed(repayToken.decimals), repayToken.decimals)`

Reply:
Alternatively, `BigNumber.from(new Big(existingLiability).mul(10 ** repayToken.decimals))`